docs/src/IconButton.doc.js

```diff
@@ -1,419 +1,862 @@
+        onDismiss={onDismiss}
+        footer={
+          <Flex alignItems="center" justifyContent="end">
+            <Button color="red" text="Save"/>
+          </Flex>
+        }
+        size="sm"
+      >
+        <Box paddingX={8}>
+          <Box marginBottom={8}>
+            <TextField
+              id="name"
+              onChange={({ value }) => console.log(value)}
+              placeholder='Like "Places to go" or "Recipes to Make"'
+              label="Name"
+              type="text"
+            />
+          </Box>
+          <Checkbox
+            checked={false}
+            id="secret"
+            label="Keep this board secret"
+            subtext="So only you and collaborators can see it."
+            name="languages"
+            onChange={({ checked }) => {
+              console.log(checked);
+            }}
           />
-        </Tooltip>
-      </Flex>
-      <Flex gap={2}>
-        <Switch
-          onChange={() => setDisabled(!disabled)}
-          id="disable-buttons"
-          switched={disabled}
-        />
-        <Box paddingX={2} flex="grow">
-          <Label htmlFor="disable-buttons">
-            <Text>Disable buttons</Text>
-          </Label>
         </Box>
-      </Flex>
-      <Flex gap={2}>
-        <Switch
-          onChange={() => setTabIndex(!tabIndex)}
-          id="unreachable-buttons"
-          switched={tabIndex}
-        />
-        <Box paddingX={2} flex="grow">
-          <Label htmlFor="unreachable-buttons">
-            <Text>Remove from keyboard navigation with tabIndex</Text>
-          </Label>
-        </Box>
-      </Flex>
-    </Flex>
+      </Modal>
+    );
+  };
+
+  const [shouldShow, setShouldShow] = React.useState(false);
+  const HEADER_ZINDEX = new FixedZIndex(10);
+  const modalZIndex = new CompositeZIndex([HEADER_ZINDEX]);
+
+  return (
+    <React.Fragment>
+      <IconButton
+        accessibilityLabel="Open edit modal"
+        icon="edit"
+        onClick={() => setShouldShow(true)}
+        size="lg"
+      />
+      {shouldShow && (
+        <Layer zIndex={modalZIndex}>
+          <ModalWithHeading onDismiss={() => setShouldShow(false)} />
+        </Layer>
+      )}
+    </React.Fragment>
   );
 }
 `}
-  />,
-);
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        description="Pair IconButton with a regular button to perform a high-emphasis action. IconButton should be a secondary action among regular buttons. "
+        defaultCode={`
+<Flex gap={2}>
+  <Button text="Cancel" size="lg"/>
+  <IconButton
+    accessibilityLabel="Open edit modal"
+    icon="trash-can"
+    onClick={() => {}}
+    iconColor="red"
+    size="lg"
+  />
+</Flex>
+`}
+      />
 
-card(
-  <Combination id="size" name="Size" size={['xs', 'sm', 'md', 'lg', 'xl']}>
-    {(props) => (
-      <IconButton accessibilityLabel={`Size ${props.color}`} icon="angled-pin" {...props} />
-    )}
-  </Combination>,
-);
+      <MainSection.Card
```

Comment:
Could you add tooltips for these IconButtons? I think more often than not we should show tooltips

Reply:
DONE